src/vm/eventpipeprovider.cpp

```diff
@@ -197,6 +197,30 @@ void EventPipeProvider::InvokeCallback()
     }
     CONTRACTL_END;
 
+    const unsigned int MaxNumberOfBytes = 1024;
+    WCHAR buffer[MaxNumberOfBytes / sizeof(WCHAR)]{};
+    EventFilterDescriptor eventFilterDescriptor{};
+    bool isEventFilterDescriptorInitialized = false;
+
+    if (pFilterData != NULL)
+    {
+        // Why can't we use standard C++ algorithms and containers?
+        const size_t filterDataLength = wcslen(pFilterData) + 1;
+        const size_t filterDataSizeInBytes = filterDataLength * sizeof(WCHAR);
+        const size_t nCharactersToCopy = filterDataSizeInBytes <= sizeof(buffer) ?
+            filterDataLength : sizeof(buffer) / sizeof(WCHAR);
+        wcscpy_s(buffer, nCharactersToCopy, pFilterData);
+
+        for (size_t i = 0; i < nCharactersToCopy; ++i)
+            if (buffer[i] == L'=' || buffer[i] == L';')
+                buffer[i] = L'\0';
+
+        eventFilterDescriptor.Ptr = (ULONGLONG)&buffer[0];
```

Comment:
You don't need to copy the pFilterData, just point at it (if this came from managed, it was already copied or pinned, so it is fine to point at it here).    This avoids the 1024 byte limitation and removes a bunch of code.

Reply:
Should I modify the original data then? Where should I replace delimiters?

---
In reply to: [230887756](https://github.com/dotnet/coreclr/pull/20796#discussion_r230887756) [](ancestors = 230887756)

Reply:
Sorry, I missed that this is also a conversion.  Perhaps just do the conversion when you do the copy in eventpipesession.cpp?

Reply:
Sure, I think that would be fine.

Reply:
There is a data transformation here (few lines above). I'm not just passing data here. Initial data is a null terminated string, but the data received by the callback is treated as a stream of key-value pairs separated by `\0`

---
In reply to: [230855829](https://github.com/dotnet/coreclr/pull/20796#discussion_r230855829) [](ancestors = 230855829)

Reply:
@jorive I missed the fact that you were transforming = and ; to null chars.    I toyed with alternatives, but decided that transforming it here is as good as the alternatives.   I would prefer to avoid propagating the wacky null character separation convention any further than where it is right now.  

It is better if you do the string manipulation with SString,  I don't know how convinient the API is for this work however. 

If that turns out to be a pain or you time out use CQuickBytes.    Roughly you do 
```
    CQuickBytes paramBytes;
    paramBytes.AllocThrows((wcslen(params) + 1) * sizeof(wchar_t));
    wchar_t* ptr = (wchar_t*) paramBytes.Ptr();
```
The buffer dies when CQuickBytes dies.

Reply:
I'm trying this approach with `SString`.

---
In reply to: [230947069](https://github.com/dotnet/coreclr/pull/20796#discussion_r230947069) [](ancestors = 230947069)